gitignore_parser.py

```diff
@@ -13,18 +13,19 @@ def handle_negation(file_path, rules: Reversible["IgnoreRule"]):
     return False
 
 def parse_gitignore(full_path, base_dir=None):
+    with open(full_path) as ignore_file:
+        return parse_gitignore_string(ignore_file.read(), full_path, base_dir)
+
+def parse_gitignore_string(gitignore_str, full_path, base_dir=None):
```

Comment:
Let's name it `parse_gitignore_str` please. (I understand it was called `_string` in the test.)